### PR TITLE
Feature/typed values

### PR DESCRIPTION
**Pull request by dabeeeenster**

### CODE REVIEW

**Comment by dabeeeenster** [comment]
@richardpiazza - when reviewing this, I noticed I couldn't build the sample project (FlagsmithClient.xcworkspace) until the TypedValue class was added to the project under Pods -> FlagsmithClient ?

**Comment by richardpiazza** [comment]
Ok... finished updating my feature branch (along with sample project & unit tests). Here's the high level changes:

* `UnknownTypeValue` becomes `TypedValue`: This includes the support for `Bool` types
* `Flag.value` & `Trait.typedValue` are a non-optional `TypeValue`: Values are always expected, but `null` is still valid.
* `getFeatureValue(withID:forIdentity:completion:)` is deprecated in favor of `getValueForFeature(withID:forIdentity:completion:)` which returns the `TypedValue` in place of only a string.

**Comment by dabeeeenster** [comment]
This is awesome thanks. Is there an easy/non painful way of running these tests as part of a GH action?

**Comment by richardpiazza** [comment]
Yep... since it's a swift package, you can use the command `swift test` to run the unit tests... Here's the yml that I use on my own projects:

```yml
name: Swift

on:
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]

jobs:
  macos-build:

    runs-on: macos-latest

    steps:
    - uses: actions/checkout@v2
    - name: Build (macOS)
      run: swift build -v
    - name: Run tests
      run: swift test -v
  
  ubuntu-build:

    runs-on: ubuntu-latest

    steps:
    - uses: actions/checkout@v2
    - name: Build (Ubuntu)
      run: swift build -v
    - name: Run tests
      run: swift test -v
```

File: FlagsmithClient/Classes/Trait.swift [hunk 1]
```diff
@@ -14,20 +14,94 @@ public struct Trait: Codable {
...
+  var value: String {
+    get { typedValue.description }
+    set { typedValue = .string(newValue) }
   }
 }
 
 /**
 A PostTrait represents a structure to set a new trait, with the Trait fields and the identity.
 */
+@available(*, deprecated)
```
Comment (richardpiazza) [inline FlagsmithClient/Classes/Trait.swift:104]: With the encoding changes I made to `Trait`, this struct is no longer used in the `APIManager`. However I left `Trait.identifer` and the corresponding initializer `Trait(trait: Trait, identifier: String)` as **internal**. Was that the original intent of this struct, to only be used internally to the module? Or should the internal initializer be made public with additional documentation as to its intended usage and encoding behavior?

**Pull request merged**